### PR TITLE
[MB-2258] iOS Badging Support

### PR DESCRIPTION
**Pull request by gatlinhebert**
Adds the ability to set and get iOS badge counts.

### CODE REVIEW

File: js/UrbanAirship.js [hunk 1]
```diff
@@ -402,6 +402,24 @@ class UrbanAirship {
...
     return UrbanAirshipModule.isUserNotificationsEnabled();
   }
+
+  /**
+   * Sets the badge number for iOS.
+   *
+   * @param {number} badgeNumber specified badge to set.
+   */
+  static setBadgeNumber(badgeNumber: number) {
+    UrbanAirshipModule.setBadgeNumber(badgeNumber);
```
Comment (rlepinski) [inline js/UrbanAirship.js]: This will fail on Android. Should we call wrap it in a platform check?

**Comment by linkmh** [comment]
:shipit:

File: js/UrbanAirship.js [hunk 2]
```diff
@@ -402,6 +402,24 @@ class UrbanAirship {
...
+  }
+
+  /**
+   * Gets the current badge number for iOS.
+   *
+   * @return {Promise.<number>} A promise with the result.
+   */
+  static getBadgeNumber(): Promise<number> {
+    return UrbanAirshipModule.getBadgeNumber();
```
Comment (rlepinski) [inline js/UrbanAirship.js]: Should we implement this method on Android and return 0 everytime?

File: js/UrbanAirship.js (hunk 1 above)
Comment (gatlinhebert) [inline js/UrbanAirship.js]: Would the following make sense? 
```
if (Platform.OS == 'ios') {
      UrbanAirshipModule.setBadgeNumber(badgeNumber);
} else {
      console.log('This feature is not supported on this platform.')
 }
```
Or is this something that should be handled within app code? 

```
if (Platform.OS == 'ios') {
  UrbanAirship.setBadgeNumber(1);
}
```

**Comment by rlepinski** [comment]
👍

**Pull request merged**